src/sentry/integrations/utils/code_mapping.py

```diff
@@ -322,21 +292,36 @@ def _generate_code_mapping_from_tree(
             for src_path in repo_tree.files
             if self._potential_match(src_path, frame_filename)
         ]
+        # It is too risky generating code mappings when there's more
+        # than one file potentially matching
         if len(matched_files) != 1:
             return []
 
-        stacktrace_root, source_path = find_roots(frame_filename.raw_path, matched_files[0])
-        # It is too risky generating code mappings when there's more
-        # than one file potentially matching
+        stack_path = frame_filename.raw_path
+        source_path = matched_files[0]
+        stack_root, source_root = find_roots(stack_path, source_path)
+
+        if stack_path.replace(stack_root, source_root, 1) != source_path:
+            logger.info(
+                "Unexpected stack_path/source_path found. A code mapping was not generated.",
+                extra={
+                    "stack_path": stack_path,
+                    "source_path": source_path,
+                    "stack_root": stack_root,
+                    "source_root": source_root,
+                },
+            )
+            return []
```

Comment:
This is a safety check for a generated code mapping that mimics the end behavior - it replaces the generated **stack root** in the **stack path** with the generated **source root**, and ensures if that is equal to the **source path**. If it isn't, then the stack path or source path must be some weird edge case that we did not consider. We log this so we will have visibility into these edge cases, so we can account for them later on. 

It's clear to me now that there are are too many weird edge cases for a stack trace file path (like urls somehow ending up in there??), and trying to capture all of them will result in an infinite game of whack-a-mole. 

Even though `find_roots` is more robust than our previous solution, this added check will hopefully help safeguard against any edge cases we didn't account for.